plugin/plugins/file/service/service.go

```diff
@@ -17,24 +17,45 @@ import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
-var _ baseapp.StreamingService = &StreamingService{}
+/*
+The naming schema and data format for the files this service writes out to is as such:
 
-// StreamingService is a concrete implementation of StreamingService that writes state changes out to files
-type StreamingService struct {
+After every `BeginBlock` request a new file is created with the name `block-{N}-begin`, where N is the block number. All
+subsequent state changes are written out to this file until the first `DeliverTx` request is received. At the head of these files,
+the length-prefixed protobuf encoded `BeginBlock` request is written, and the response is written at the tail.
+
+After every `DeliverTx` request a new file is created with the name `block-{N}-tx-{M}` where N is the block number and M
```

Comment:
question: would this not cause an extreme amount of files in the file system? would it not be better to have one file for all txs/state updates in a block?